src/main/java/com/google/sps/data/Utility.java

```diff
@@ -61,179 +59,39 @@ public static String graphToJson(MutableGraph<GraphNode> graph, HashSet<String>
-          graphNodesMap.put(startName, newGraphNode);
-        }
-        break;
-      case ADD_EDGE:
-        if (startNode == null || endNode == null) { // Check nodes exist before adding an edge
-          return false;
-        }
-        // The target cannot be a root since it has an in-edge
-        roots.remove(endName);
-        graph.putEdge(startNode, endNode);
-        break;
-      case DELETE_NODE:
-        if (startNode == null) { // Check node exists before removing
-          return false;
-        }
-        // Check whether any successor will have no in-edges after this node is removed
-        // If so, make them roots
-        Set<GraphNode> successors = graph.successors(startNode);
-        for (GraphNode succ : successors) {
-          if (graph.inDegree(succ) == 1) {
-            roots.add(succ.name());
-          }
-        }
-        roots.remove(startName);
-        graph.removeNode(startNode); // This will remove all edges associated with startNode
-        graphNodesMap.remove(startName);
-        break;
-      case DELETE_EDGE:
-        if (startNode == null || endNode == null) { // Check nodes exist before removing edge
-          return false;
-        }
-        graph.removeEdge(startNode, endNode);
-        // If the target now has no in-edges, it becomes a root
-        if (graph.inDegree(endNode) == 0) {
-          roots.add(endName);
-        }
-        break;
-      case CHANGE_TOKEN:
-        if (startNode == null) {
-          return false;
-        }
-        return changeNodeToken(startNode, mut.getTokenChange());
-      default:
-        // unrecognized mutation type
-        return false;
-    }
-    return true;
-  }
-
-  /*
-   * Modify the list of tokens for graph node 'node' to accomodate the mutation
-   * 'tokenMut'. This could involve adding or removing tokens from the list.
-   *
-   * @param node the node in the graph to change the tokens of
-   * @param tokenMut the kind of mutation to perform on node of the graph
+  /**
+   * Returns a mutable shallow copy of the given guava graph
    *
-   * @return true if the change is successful, false otherwise
+   * @param graph the guava graph to shallow copy
+   * @return a shallow copy of the given guava graph
    */
-  private static boolean changeNodeToken(GraphNode node, TokenMutation tokenMut) {
-    // List of tokens to add/remove from the existing list
-    List<String> tokenNames = tokenMut.getTokenNameList();
-    // The existing list of tokens in the node
-    List<String> tokenList = node.tokenList();
-    TokenMutation.Type tokenMutType = tokenMut.getType();
-    if (tokenMutType == TokenMutation.Type.ADD_TOKEN) {
-      tokenList.addAll(tokenNames);
-    } else if (tokenMutType == TokenMutation.Type.DELETE_TOKEN) {
-      tokenList.removeAll(tokenNames);
-    } else {
-      // unrecognized mutation
-      return false;
-    }
-    return true;
+  public static MutableGraph<GraphNode> copyGraph(MutableGraph graph) {
```

Comment:
Is there really a point of this function? It seems straightforward enough to say Graphs.copyOf(graph) in DataGraph. I really don't think this is needed?

Reply:
Yep removed it!